flocker/control/test/test_persistence.py

```diff
@@ -249,3 +252,32 @@ def test_complex_keys(self):
                                manifestations={}, paths={},
                                devices={uuid4(): FilePath(b"/tmp")})
         self.assertEqual(node_state, wire_decode(wire_encode(node_state)))
+
+
+class ConfigurationMigrationTests(SynchronousTestCase):
+    """
+    Tests for ``_ConfigurationMigration``.
+    """
+    configurations_dir = FilePath(__file__).sibling('configurations')
+
+    def test_v0_v1_configuration(self):
+        """
+        A V0 JSON configuration blob is transformed to a V1 configuration
+        blob, with the result validating when loaded.
+        """
+        v0_json = self.configurations_dir.child(
+            'configuration_v0.json').getContent()
+        v1_json = migrate_configuration(0, 1, v0_json)
+        v1_config = wire_decode(v1_json)
+        self.assertEqual(v1_config, Deployment(nodes=frozenset()))
+
+    def test_v1_v0_configuration(self):
+        """
+        A V1 JSON configuration blob is transformed to a V0 configuration
+        blob.
+        """
+        config_dict = {_CLASS_MARKER: u"Deployment"}
```

Comment:
:question: is it not plausible that someone might run a v2 config from a later version of Flocker and then at some point need to revert Flocker to an older version that doesn't support v2? I'm happy to skip downgrades (it means I do less work on this branch, after all), but I've always been of the view that model migrations in a system should be bidirectional.